packages/core/server.py

```
@@ -116,18 +138,26 @@ def run_skill(self, name: str, args: Dict[str, Any]) -> Dict[str, Any]:
             try:
-                return json.loads(output)
+                result = json.loads(output)
+                logger.info("Skill executed successfully", extra={"skill_name": name})
+                return result
             except:
+                logger.warning("Skill output is not JSON", extra={"skill_name": name})
                 return {"raw_output": output}
 
         except subprocess.TimeoutExpired:
-            return {"error": f"Skill execution timed out after {timeout}s"}
+            logger.error("Skill execution timed out", extra={"skill_name": name, "timeout": timeout})
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)